hack/generated/controllers/controller_resources.go

```diff
@@ -5,30 +5,23 @@ Licensed under the MIT license.
 
 package controllers
 
-// TODO: Code generate this file
 import (
 	"k8s.io/apimachinery/pkg/runtime"
 
-	batch "github.com/Azure/k8s-infra/hack/generated/_apis/microsoft.batch/v20170901"
-	documentdb "github.com/Azure/k8s-infra/hack/generated/_apis/microsoft.documentdb/v20150408"
 	resources "github.com/Azure/k8s-infra/hack/generated/_apis/microsoft.resources/v20200601"
-	storage "github.com/Azure/k8s-infra/hack/generated/_apis/microsoft.storage/v20190401"
 )
 
-// KnownTypes defines an array of runtime.Objects to be reconciled, where each
-// object in the array will generate a controller. If the concrete type
-// implements the owner interface, the generated controllers will inject the
-// owned types supplied by the Owns() method of the CRD type. Each controllers
-// may directly reconcile a single object, but may indirectly watch
-// and reconcile many Owned objects. The singular type is necessary to generically
-// produce a reconcile function aware of concrete types, as a closure.
-
-var KnownTypes = []runtime.Object{
-	new(batch.BatchAccount),
-	// new(batch.BatchAccountsPool),
-	new(documentdb.DatabaseAccount),
-	new(resources.ResourceGroup),
-	new(storage.StorageAccount),
-	new(storage.StorageAccountsBlobService),
-	new(storage.StorageAccountsBlobServicesContainer),
+func GetKnownTypes() []runtime.Object {
```

Comment:
Would it be clearer to change the generators for known types and schemes to always add resources? I guess it's much of a muchness. (Not sure whether you have that expression in the US?)

Reply:
No won't don't have that phrase in the US. I looked it up though and it sound sorta similar to "6 of one, half-dozen of the other"?

I'm not actually sure what you're getting at here ("change the generators for known types and schemes to always add resources"). Can you explain more?

Reply:
Oh sorry! Yes, "6 of one, half-a-dozen of the other" is the same sense. I just meant, rather than generate private `getKnownTypes` and then wrap a `GetKnownTypes` around it to add the resource type, it might be better to push the code that adds it into the generator? Or might not.

Reply:
Yeah my thinking was that it was easier to add other logic if the wrapper was handcrafted (as then it doesn't require changing the generator). So the generated code is the "building block" and then the handcrafted wrapper makes use of that building block.

Easy enough to change down the road if we determine we don't like that but I think I'll leave it for now.